docs/usage.rst

```diff
@@ -6,12 +6,61 @@ Usage
 
 .. sectnum:: :start: 1
 
-Any map can be completely specified by two objects, a numpy array (of at least two dimensions) whose two trailing dimensions correspond to two coordinate axes of the map, and a ``wcs`` object that specifies the World Coordinate System. The latter specifies the correspondence between pixels and physical sky coordinates. This library allows for the manipulation of an object ``ndmap``, which has all the properties of numpy arrays but is in addition enriched by the ``wcs`` object (specifically an instantiation of Astropy's ``astropy.wcs.wcs.WCS`` object). The ``shape`` of the numpy array and the ``wcs`` completely specifies the geometry and footprint of a map of the sky.
-
-All ``ndmap`` s must have at least two dimensions. The trailing two axes are interpreted as the Y (typically, declination) and X (typically, right ascension) axes. Maps can have arbitrary number of leading dimensions, but many of ``pixell``' CMB-related tools interpret a 3D array of shape ``(ncomp,Ny,Nx)`` to consist of three ``Ny`` x ``Nx`` maps of intensity, polarization Q and U Stokes parameters in that order.
+The ``ndmap`` object
+--------------------
+
+Any map can be completely specified by two objects, a numpy array (of
+at least two dimensions) whose two trailing dimensions correspond to
+two coordinate axes of the map, and a ``wcs`` object that specifies
+the World Coordinate System. The latter specifies the correspondence
+between pixels and physical sky coordinates. This library allows for
+the manipulation of an object ``ndmap``, which has all the properties
+of numpy arrays but is in addition enriched by the ``wcs`` object
+(specifically an instantiation of Astropy's ``astropy.wcs.wcs.WCS``
+object). The combination of the ``shape`` of the numpy array and the
+``wcs`` completely specifies the footprint of a map of the sky, and is
+called the ``geometry``.
+
+An ``ndmap`` must have at least two dimensions. The two right-most
+axes represent celestial coordinates (typically Declination and Right
+Ascension).  Maps can have arbitrary number of leading dimensions, but
+many of the ``pixell`` CMB-related tools interpret 3D arrays with
+shape ``(ncomp,Ny,Nx)`` as representing ``Ny`` x ``Nx`` maps of
+intensity, polarization Q and U Stokes parameters, in that order.
+
+In the two celestial axes, the "first" pixel (index [0,0]) corresponds
+to the lower left-hand corner of the map, as it would be displayed on
+a screen or printed page.  The first index represents the row,
+parallel to the Cartesian Y-axis, typically associated with increasing
+Declination.  The second index represents the column, parallel to the
+Cartesian X-axis, typically associated with decreasing Right
+Ascension.
```

Comment:
I think these statements are WCS dependent right? esp. "In the two celestial axes, the "first" pixel (index [0,0]) corresponds to the lower left-hand corner of the map, as it would be displayed on
a screen or printed page." is not very meaningful since the array itself can be stored in memory any way one wants (depending on e.g. sign of cdelt). Similarly "typically" in the last sentence also needs [citation] (I never remember which way things are typically stored since I've been using pixell/enlib way too long :) ).

Reply:
From FITS-I (at link below, see section 5.1), "we recommend that FITS writers order the pixels so that the first pixel in the FITS file (for each image plane) be the one that would be displayed in the lower-left corner (with the first axis increasing to the right and the second axis increasing upwards)".

I'll rewrite this to not sound so definitive... but can we "strongly recommend" this convention?  The reasons are explained below.  To my knowledge, existing standard ACT headers follow this convention.



There's a good reason for such a convention -- it answers the question of "how should I display this image, on a screen".  If you already have a strong preference for this (e.g., "I want to represent the full equatorial sky with a CAR-like projection on my monitor"), then that's fine, your code can simply transfer the numbers in the data into the appropriate pixels on your screen.  But that choice (CAR on the full sky) won't always work.  Some data live near a pole, and they won't look good on your CAR projection.  Some maps might have pixel axes that run at 45 degrees to your lines of constant RA, and those will be distorted in your reprojection.

So it's reasonable to ask "in the absence of me having a personal preference for how to project these data onto my monitor, how should I display the data?"  Did the file writer tell you this?  For one thing, the file writer has used a grid to represent the data.  So that's helpful, given that monitors also have pixel grids.  But there are 8 different ways you could align this pair of 2-d grids.  I concede that you can, generally, eliminate half of those by asking that RA and Dec have the traditional handedness... and then you could decide between the remaining options if you were working with standard special cases, such as the usual ACT CAR/CEA.

In contrast a general image display code is easily developed if you are following the lower-left convention.  First, you draw the pixels on the screen (starting in the lower left...).  Then you check if the WCS is within one of your well-understood classes (such as simple versions of CEA/CAR) where the coordinates can be entirely communicated through X and Y axis tick labels.  If so, tick those axes and label those ticks.  If not, overlay lines of constant RA and Dec on your images, and label those somehow.

Refs: https://www.atnf.csiro.au/people/mcalabre/WCS/, esp. papers I and II.

Reply:
Note that while pixell.enmap doesn't care what the WCS is, pixell.enplot makes sure that the map is plotted in the convention you refer to, regardless of what the convention in the WCS was. We should definitely expose enplot more (there's a bunch of docstrings but not much else) and encourage people to use it. I was initially hesitant to have the readthedocs be too astronomy-centric, but I think this is a losing battle because pixell already is pretty astronomy-centric, and I'm ok with the docs specifying and encouraging good astronomy-oriented FITS-convention practices. So feel free to add those.